src/Service.Tests/GraphQLBuilder/NestedMutationBuilderTests.cs

```diff
@@ -0,0 +1,408 @@
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Resolvers;
+using Azure.DataApiBuilder.Core.Resolvers.Factories;
+using Azure.DataApiBuilder.Core.Services;
+using Azure.DataApiBuilder.Core.Services.MetadataProviders;
+using Azure.DataApiBuilder.Core.Models;
+using HotChocolate.Language;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Azure.DataApiBuilder.Core.Services.Cache;
+using ZiggyCreatures.Caching.Fusion;
+using Azure.DataApiBuilder.Core.Authorization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using static Azure.DataApiBuilder.Service.GraphQLBuilder.GraphQLNaming;
+using System.Linq;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Directives;
+using Azure.DataApiBuilder.Service.GraphQLBuilder.Mutations;
+using Azure.DataApiBuilder.Service.GraphQLBuilder;
+
+namespace Azure.DataApiBuilder.Service.Tests.GraphQLBuilder
+{
+    /// <summary>
+    /// Parent class containing tests to validate different aspects of schema generation for nested mutations for different type of
+    /// relational database flavours supported by DAB.
+    /// </summary>
+    [TestClass]
+    public abstract class NestedMutationBuilderTests
+    {
+        // Stores the type of database - MsSql, MySql, PgSql, DwSql. Currently nested mutations are only supported for MsSql.
+        protected static string databaseEngine;
+
+        // Stores mutation definitions for entities.
+        private static IEnumerable<IHasName> _mutationDefinitions;
+
+        // Stores object definitions for entities.
+        private static IEnumerable<IHasName> _objectDefinitions;
+
+        // Runtime config instance.
+        private static RuntimeConfig _runtimeConfig;
+
+        #region Nested Create tests
+
+        /// <summary>
+        /// Test to validate that we don't expose the object definitions inferred for linking entity/table to the end user as that is an information
+        /// leak. These linking object definitions are only used to generate the final source->target linking object definitions for entities
+        /// having an M:N relationship between them.
+        /// </summary>
+        /// <param name="sourceEntityName">Name of the source entity for which the configuration is provided in the config.</param>
+        /// <param name="targetEntityName">Name of the target entity which is related to the source entity via a relationship defined in the 'relationships'
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author", DisplayName = "Validate absence of linking object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book", DisplayName = "Validate absence of linking object for Author->Book M:N relationship")]
+        [DataRow("BookNF", "AuthorNF", DisplayName = "Validate absence of linking object for BookNF->AuthorNF M:N relationship")]
```

Comment:
How is BookNF different than Book?

Reply:
They are two different exposed entities backed by same books table. In terms of this test, there is no difference. These are just multiple test cases.

Reply:
Are those multiple cases testing different scenarios?

Reply:
No, they are not. Yeah makes sense to have only one, I'll adjust the rest and see if they provide any additional coverage.